django/contrib/sitemaps/__init__.py

```diff
@@ -202,23 +209,25 @@ def _urls(self, page, protocol, domain):
 
             if self.i18n and self.alternates:
                 for lang_code in self._languages():
-                    loc = f"{protocol}://{domain}{self._location(item, lang_code)}"
-                    url_info["alternates"].append(
-                        {
-                            "location": loc,
-                            "lang_code": lang_code,
-                        }
-                    )
+                    if self.enable_item_for_lang(item[0], lang_code):
+                        loc = f"{protocol}://{domain}{self._location(item, lang_code)}"
+                        url_info["alternates"].append(
+                            {
+                                "location": loc,
+                                "lang_code": lang_code,
+                            }
+                        )
                 if self.x_default:
-                    lang_code = settings.LANGUAGE_CODE
-                    loc = f"{protocol}://{domain}{self._location(item, lang_code)}"
-                    loc = loc.replace(f"/{lang_code}/", "/", 1)
-                    url_info["alternates"].append(
-                        {
-                            "location": loc,
-                            "lang_code": "x-default",
-                        }
-                    )
+                    if self.enable_item_for_lang(item[0], lang_code):
+                        lang_code = settings.LANGUAGE_CODE
```

Comment:
`lang_code` should be set before the check:
```suggestion
                    lang_code = settings.LANGUAGE_CODE
                    if self.enable_item_for_lang(item[0], lang_code):
```